src/client.cpp

```diff
@@ -908,6 +905,56 @@ void CClient::Stop()
     SignalLevelMeter.Reset();
 }
 
+/// @method
+/// @brief Connects to strServerAddress
+/// @emit Connecting (strServerName) if the client wasn't running and SetServerAddr returned true.
+/// @param strServerAddress - the server address to connect to
+/// @param strServerName - the String argument to be passed to Connecting()
+/// @result true if client wasn't running and SetServerAddr returned true, false otherwise
+bool CClient::Connect ( QString strServerAddress, QString strServerName )
+{
+    if ( !IsRunning() )
+    {
+        // Set server address and connect if valid address was supplied
+        if ( SetServerAddr ( strServerAddress ) )
```

Comment:
I don't like this style at all.  If there's some validation to happen, it should have happened before we get this far.  We shouldn't _try_ to connect to an address that's not invalid.

Perhaps this whole routine is redundant and `SetServerAddr` should be called `Connect` and do the emits?

Reply:
> SetServerAddr

Should just do exactly what the name implies IMO - Otherwise we could introduce mixing of functionality.
Connect() would then emit the signals. That's cleaner and closer to what we do now.